src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,40 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH_OF_PASSWORD_AND_LOGIN = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidDataException("The user with the same login already exists");
+        }
+
+        if (storageDao.get(user.getLogin()) == null) {
+            if (user.getLogin() == null) {
+                throw new InvalidDataException("The login cannot be null");
+            }
+            if (user.getPassword() == null) {
+                throw new InvalidDataException("The password cannot be null");
+            }
+            if (String.valueOf(user.getAge()) == null) {
+                throw new InvalidDataException("The age cannot be null");
+            }
+        }
+
+        if (user.getLogin().length() < MIN_LENGTH_OF_PASSWORD_AND_LOGIN) {
+            throw new InvalidDataException("The length of login is less than 6 characters");
+        } else if (user.getPassword().length() < MIN_LENGTH_OF_PASSWORD_AND_LOGIN) {
+            throw new InvalidDataException("The length of password is less than 6 characters");
+        } else if (user.getAge() < 18) {
+            throw new InvalidDataException("The age is less than 18 years");
```

Comment:
The if-else structure here is complicated and not necessary. You should split the conditions into separate if statements without using else if. This will make the code cleaner and more readable, and it will also align with the checklist which advises against complicating if-else statements.